vllm/v1/engine/llm_engine.py

```diff
@@ -211,7 +211,7 @@ def add_request(
             # Add the request to EngineCore.
             self.engine_core.add_request(child_request)
 
-    def step(self) -> list[RequestOutput]:
+    def step(self) -> list[RequestOutput | PoolingRequestOutput]:
```

Comment:
```suggestion
    def step(self) -> list[Union[RequestOutput, PoolingRequestOutput]]:
```

Need to remain compatible with Python 3.9